pkg/secrets/passphrase/manager.go

```diff
@@ -27,7 +28,9 @@ import (
 	"sync"
 
 	"github.com/pulumi/pulumi/pkg/v3/secrets"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/diag"
 	"github.com/pulumi/pulumi/sdk/v3/go/common/resource/config"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
```

Comment:
If we want to avoid this dependency, we could define an interface like

```go
interface Prompter {
    ReadPassword(prompt string) (string, error)
}
```

Reply:
Yeah, that's basically the approach I was going down initially (although, typed as a func rather than an interface), but then you have to pass it through multiple layers. In the end, I actually don't think it's so bad having `passphrase` use `cmdutil` as it keeps all the passphrase-related code together, and doesn't require injecting the dependency and plumbing it through.